hudi-spark-datasource/hudi-spark/src/main/scala/org/apache/spark/sql/hudi/command/procedures/HoodieProcedureUtils.scala

```diff
@@ -43,4 +47,74 @@ object HoodieProcedureUtils {
 
     ProcedureArgs(isNamedArgs = true, map, new GenericInternalRow(values))
   }
+
+  sealed trait Operation {
+    def value: String
+    def isSchedule: Boolean = this != Execute
+    def isExecute: Boolean = this != Schedule
```

Comment:
maybe that define two methods, and implement them in sub-classes, is more clear.